File: backend/Models/PropertyView.cs
namespace QBooking.Models
{
    public class PropertyView
    {
        public long Id { get; set; }

        public int PropertyId { get; set; }
        public int? UserId { get; set; }

        public DateTime ViewedAt { get; set; } = DateTime.UtcNow;
        public string? IPAddress { get; set; }
        public string? UserAgent { get; set; }
        public string? Referrer { get; set; }

        // Navigation
        public Property Property { get; set; } = null!;
        public User? User { get; set; }
    }
}

File: frontend/src/types/main/property-view.ts
// types/property-view.ts

export interface PropertyViewRequest {
  propertyId: number;
}

export interface PropertyViewResponse {
  id: number;
  propertyId: number;
  userId?: number;
  viewedAt: string;
  ipAddress?: string;
  userAgent?: string;
}

export interface ApiResponse<T = any> {
  success: boolean;
  message?: string;
  data?: T;
  statusCode?: number;
}